src/web/app/components/response-time-series-chart/response-time-series-chart.component.ts

```diff
@@ -40,10 +42,29 @@ export class ResponseTimeSeriesChartComponent implements OnInit {
     this.feedbackResponseStatsService.loadResponseStats(durationSeconds.toString(),
         this.model.intervalSeconds.toString())
         .subscribe((records: FeedbackResponseRecords) => {
-          console.log(records);
-          this.drawChart(records.responseRecords, durationSeconds * 1000);
+          const canvas: any = d3.select('svg');
+
+          // clear all content
+          canvas.selectAll('*').remove();
+
+          this.drawChart(canvas, records.responseRecords, durationSeconds * 1000, false, '#696969');
+
+          const temp: number[] = [];
+          records.responseRecords.forEach((record: FeedbackResponseRecord, i: number) => {
```

Comment:
This entire thing seems very odd...
Could read better if you use a for loop instead.
I believe you mean `map` here but found forEach instead.

Reply:
forEach seems to work too, in javascript, but yeah i agree `map` would be more apt